src/lib/io/include/sol-uart.h

```diff
@@ -53,36 +53,79 @@ extern "C" {
 
 struct sol_uart;
 
-struct sol_uart *sol_uart_open(const char *port_name);
-void sol_uart_close(struct sol_uart *uart);
-
-bool sol_uart_set_baud_rate(struct sol_uart *uart, uint32_t baud_rate);
-uint32_t sol_uart_get_baud_rate(const struct sol_uart *uart);
-
-bool sol_uart_set_parity_bit(struct sol_uart *uart, bool enable, bool odd_paraty);
-bool sol_uart_get_parity_bit_enable(struct sol_uart *uart);
-bool sol_uart_get_parity_bit_odd(struct sol_uart *uart);
+enum sol_uart_baud_rate {
+    SOL_UART_BAUD_RATE_9600 = 0,
+    SOL_UART_BAUD_RATE_19200,
+    SOL_UART_BAUD_RATE_38400,
+    SOL_UART_BAUD_RATE_57600,
+    SOL_UART_BAUD_RATE_115200
+};
+
+enum sol_uart_data_bits {
+    SOL_UART_DATA_BITS_8 = 0,
+    SOL_UART_DATA_BITS_7,
+    SOL_UART_DATA_BITS_6,
+    SOL_UART_DATA_BITS_5
+};
+
+enum sol_uart_parity {
+    SOL_UART_PARITY_NONE = 0,
+    SOL_UART_PARITY_EVEN,
+    SOL_UART_PARITY_ODD
+};
+
+enum sol_uart_stop_bits {
+    SOL_UART_STOP_BITS_ONE = 0,
+    SOL_UART_STOP_BITS_TWO
+};
+
+struct sol_uart_config {
+#define SOL_UART_CONFIG_API_VERSION (1)
+    uint16_t api_version;
+    enum sol_uart_baud_rate baud_rate;
+    enum sol_uart_data_bits data_bits;
+    enum sol_uart_parity parity;
+    enum sol_uart_stop_bits stop_bits;
+    /* Set a callback to be called every time a character is received on UART */
```

Comment:
Missing doxygen token '/**<' and needs to be inline with the field (or add a struct documentation block above the struct). Same for line 92.